tests/unit/test_command_install.py

```diff
@@ -9,39 +10,90 @@
     decide_user_install,
     warn_deprecated_install_options,
 )
+from pip._internal.exceptions import CommandError, InstallationError
 from pip._internal.req.req_install import InstallRequirement
 
+ENABLE_USER_SITE = 'site.ENABLE_USER_SITE'
+WRITABLE = 'pip._internal.commands.install.test_writable_dir'
+ISDIR = 'pip._internal.commands.install.os.path.isdir'
+EXISTS = 'pip._internal.commands.install.os.path.exists'
 
+
+def false(*args, **kwargs):
+    """Return False."""
+    return False
+
+
+def true(*args, **kwargs):
+    """Return True."""
+    return True
+
+
+# virtualenv_no_global is tested by functional test
+@patch('pip._internal.commands.install.virtualenv_no_global', false)
 class TestDecideUserInstall:
-    @patch('site.ENABLE_USER_SITE', True)
-    @patch('pip._internal.commands.install.site_packages_writable')
-    def test_prefix_and_target(self, sp_writable):
-        sp_writable.return_value = False
-
-        assert decide_user_install(
-            use_user_site=None, prefix_path='foo'
-        ) is False
-
-        assert decide_user_install(
-            use_user_site=None, target_dir='bar'
-        ) is False
-
-    @pytest.mark.parametrize(
-        "enable_user_site,site_packages_writable,result", [
-            (True, True, False),
-            (True, False, True),
-            (False, True, False),
-            (False, False, False),
-        ])
-    def test_most_cases(
-        self, enable_user_site, site_packages_writable, result, monkeypatch,
-    ):
-        monkeypatch.setattr('site.ENABLE_USER_SITE', enable_user_site)
-        monkeypatch.setattr(
-            'pip._internal.commands.install.site_packages_writable',
-            lambda **kw: site_packages_writable
-        )
-        assert decide_user_install(use_user_site=None) is result
+    @pytest.mark.parametrize('use_user_site,prefix_path,target_dir,root_path',
```

Comment:
It's easier to maintain when params are separated:
```suggestion
    @pytest.mark.parametrize(('use_user_site', 'prefix_path', 'target_dir', 'root_path'),
```